src/read.jl

```diff
@@ -10,50 +10,45 @@ function read_case(filepath)
     case = Dict()
 
     # AC branches
-    # dropdims() will remove extraneous dimension
-    case["branchid"] = dropdims(mpc["branchid"]; dims=2)
-    # convert() will convert float array to int array
-    case["branch_from"] = convert(Array{Int,1}, mpc["branch"][:, 1])
-    case["branch_to"] = convert(Array{Int,1}, mpc["branch"][:, 2])
-    case["branch_reactance"] = mpc["branch"][:, 4]
-    case["branch_rating"] = mpc["branch"][:, 6]
+    branch = CSV.File(joinpath(filepath, "branch.csv"))
+    case["branchid"] = convert(Array{Int,1}, branch.branch_id)
+    case["branch_from"] = convert(Array{Int,1}, branch.from_bus_id)
+    case["branch_to"] = convert(Array{Int,1}, branch.to_bus_id)
+    case["branch_reactance"] = convert(Array{Float64,1}, branch.x)
+    case["branch_rating"] = convert(Array{Float64,1}, branch.rateA)
 
     # DC branches
-    if "dcline" in keys(mpc)
-        if isa(mpc["dclineid"], Int)
-            case["dclineid"] = Int64[mpc["dclineid"]]
-        else
-            case["dclineid"] = dropdims(mpc["dclineid"]; dims=2)
-        end
-        case["dcline_from"] = convert(Array{Int,1}, mpc["dcline"][:, 1])
-        case["dcline_to"] = convert(Array{Int,1}, mpc["dcline"][:, 2])
-        case["dcline_pmin"] = mpc["dcline"][:, 10]
-        case["dcline_pmax"] = mpc["dcline"][:, 11]
-    else
-        case["dclineid"] = Int64[]
-        case["dcline_from"] = Int64[]
-        case["dcline_to"] = Int64[]
-        case["dcline_pmin"] = Float64[]
-        case["dcline_pmax"] = Float64[]
-    end
-
+    dcline = CSV.File(joinpath(filepath, "dcline.csv"))
+    case["dclineid"] = convert(Array{Int,1}, dcline.dcline_id)
+    case["dcline_from"] = convert(Array{Int,1}, dcline.from_bus_id)
+    case["dcline_to"] = convert(Array{Int,1}, dcline.to_bus_id)
+    case["dcline_pmin"] = convert(Array{Float64,1}, dcline.Pmin)
+    case["dcline_pmax"] = convert(Array{Float64,1}, dcline.Pmax)
+    
     # Buses
-    case["busid"] = convert(Array{Int,1}, mpc["bus"][:, 1])
-    case["bus_demand"] = mpc["bus"][:, 3]
-    case["bus_zone"] = convert(Array{Int,1}, mpc["bus"][:, 7])
+    bus = CSV.File(joinpath(filepath, "bus.csv"))
+    case["busid"] = convert(Array{Int,1}, bus.bus_id)
+    case["bus_demand"] = convert(Array{Float64,1}, bus.Pd)
+    case["bus_zone"] = convert(Array{Int,1}, bus.zone_id)
+    try
+        case["bus_eiaid"] = convert(Array{Int,1}, mpc["bus"][:, 18])
```

Comment:
Instead of using an index number (i.e., 18), use the name of the column. So this line would be something like `case["bus_eiaid"] = convert(Array{Int,1}, bus.bus_eiaid)` (replace "bus_eiaid" with the correct column name if it is different)

Reply:
> Instead of using an index number (i.e., 18), use the name of the column. So this line would be something like `case["bus_eiaid"] = convert(Array{Int,1}, bus.bus_eiaid)` (replace "bus_eiaid" with the correct column name if it is different)

I have changed the column name and rebased just now. Please check if the re-basing is working properly. I will start working on PowerSimData while this is underway.